src/main/scala/is/hail/rvd/RVDContext.scala

```diff
@@ -0,0 +1,27 @@
+package is.hail.rvd
+
+import is.hail.annotations.Region
+
+object RVDContext {
+  def default: RVDContext = SimpleRVDContext(Region())
+
+  def fromRegion(region: Region): RVDContext = SimpleRVDContext(region)
+}
+
+// NB: must be *Auto*Closeable because calling close twice is undefined behavior
+// (see AutoCloseable javadoc)
+trait RVDContext extends AutoCloseable {
```

Comment:
I don't think this should be abstract.  I mean, it doesn't hurt, but I think it signals the wrong thing about the design.  RVDContext needs to be universal in a sense: it needs to work for all RVD operations, everywhere, and it is created by the leaf nodes of the RVD computational graph.  How could some internal node that needs a specialized version of RVDContext every expect to be arrange for every leaf node that might feed it values to produce the specialized version?  What if two different operations need two specialized RVDContexts?